app/groups/[session]/page.tsx

```diff
@@ -0,0 +1,288 @@
+'use client'
+
+import '../../i18n';
+import { Header } from "@/app/components/dashboard/header";
+import Container from '@mui/material/Container';
+import { Box, Button, FormControl, OutlinedInput } from "@mui/material";
+import { useTheme } from "@mui/material/styles";
+import AddIcon from '@mui/icons-material/Add';
+import CircularProgress from '@mui/material/CircularProgress';
+import Typography from '@mui/material/Typography';
+import { DataGrid } from '@mui/x-data-grid';
+import React, { FormEvent, useEffect, useState } from 'react';
+import ConfirmationDialog from '../../components/ConfirmationDialog';
+import { useTranslation } from "react-i18next";
+import Establishment from '@/app/components/establishment/Establishment';
+import { Session } from '@/app/types/Session';
+import { styled } from "@mui/system";
+import { archiveStudentSession, createSession, getSessionsStudents } from '@/api/sessions';
+import { getUserCookies } from '@/api/auth';
+import { useParams } from 'next/navigation'
+type Params = {
+    session: string;
+}
+
+const CustomContainer = styled('div')`
+    position: fixed;
+    top: 0;
+    left: 0;
+    right: 0;
+    z-index: 1030;
+    background-color: white;
+`
+
+const AccueilWrapper = styled(Box)`
+    margin-top: 150px;
+    a { color: #6d6d6d; }
+    a:hover { color: black; }
+`
+export default function Sessions() {
+    const theme = useTheme();
+
+    const [currentSession, setCurrentSessions] = useState<Session[]>([]);
+    const [sessions, setSessions] = useState<Session[]>([]);
+    const [loading, setLoading] = useState(false);
+    const { t } = useTranslation();
+
+    const deleteCurrentSession = (s: Session) => {
+        archiveStudentSession(s)
+        setCurrentSessions(currentSession.filter((e: Session) => s.id != e.id));
+    }
+
+    const deleteSession = (session: Session) => {
+        setSessions(sessions.filter((s: Session) => session.id != s.id));
```

Comment:
Attention, quand tu utilises la valeur précédente dans un useState il faut que tu utilises la syntaxe `setSessions(prevSession => prevSession.filter..)` 

Le setState n'est pas synchrone, quand tu passes à la ligne d'après la valeur n'a pas forcément directement changé. Et donc ici tu n'es pas sure d'avoir la dernière valeur de session. 

cf https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state